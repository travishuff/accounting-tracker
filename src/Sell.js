import React from 'react'
import axios from 'axios'
import { Button, Form, FormInput, FormGroup } from 'shards-react'

class Sell extends React.Component {
  state = {
    number: '',
    sellDate: '',
  }

  handleNumberChange = event => {
    if (!event.target.value) {
      this.setState({
        number: '',
      })
    } else {
      this.setState({
        number: Number(event.target.value),
      })
    }
  }
  handleChange = event => {
    this.setState({
      [event.target.name]: event.target.value,
    })
  }
  handleSubmit = async event => {
    event.preventDefault()

    try {
      const response = await axios.put(
        'http://localhost:8080/api/bananas',
        this.state
      )
      console.log('submitted!', response)
    } catch (error) {
      console.error(error)
    }
  }
  handleReset = () => {
    this.setState({
      number: '',
      sellDate: '',
    })
  }
  onSubmit = event => {
    this.handleSubmit(event)
      .then(this.handleReset())
      .catch(console.error)
  }

  render() {
    const { number, sellDate } = this.state

    return (
      <div>
        <div>Sell</div>
        <Form onSubmit={this.onSubmit}>
          <FormGroup>
            <label htmlFor="number">
              Number of bananas to sell
              <FormInput
                size="sm"
                type="number"
                name="number"
                value={number}
                onChange={this.handleNumberChange}
                onBlur={this.handleNumberChange}
              />
            </label>
          </FormGroup>
          <FormGroup>
            <label htmlFor="sellDate">
              Sale date
              <FormInput
                size="sm"
                type="date"
                name="sellDate"
                value={sellDate}
                onChange={this.handleChange}
                onBlur={this.handleChange}
              />
            </label>
          </FormGroup>
          <Button size="sm">Submit</Button>
          <Button size="sm" theme="danger" onClick={this.handleReset}>
            Reset
          </Button>
        </Form>
      </div>
    )
  }
}

export default Sell
