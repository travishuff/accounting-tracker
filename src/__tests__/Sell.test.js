import React from 'react'
import { create } from 'react-test-renderer'

import Sell from '../Sell'

// Snapshot test
test('snapshot', () => {
  const c = create(<Sell />)

  expect(c.toJSON()).toMatchSnapshot()
})

// Unit testing handleReset method
test('handleReset method resets state', () => {
  const c = create(<Sell />)
  const instance = c.getInstance()

  instance.state.number = 22
  instance.state.sellDate = '2019-04-13'

  expect(instance.state.number).toBe(22)
  expect(instance.state.sellDate).toBe('2019-04-13')
  instance.handleReset()
  expect(instance.state.number).toBe('')
  expect(instance.state.sellDate).toBe('')
})
