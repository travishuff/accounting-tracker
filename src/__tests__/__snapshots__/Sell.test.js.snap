// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row-title row"
  >
    <div
      className="col"
    >
      Sell Bananas
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <form
        className=""
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="number"
          >
            Number of bananas to sell:
            <input
              className="form-control form-control-sm"
              name="number"
              onBlur={[Function]}
              onChange={[Function]}
              type="number"
              value=""
            />
          </label>
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="sellDate"
          >
            Sale date:
            <input
              className="form-control form-control-sm"
              name="sellDate"
              onBlur={[Function]}
              onChange={[Function]}
              type="date"
              value=""
            />
          </label>
        </div>
        <button
          className="btn btn-primary btn-sm"
          onClick={[Function]}
        >
          Submit
        </button>
        <button
          className="btn btn-danger btn-sm"
          onClick={[Function]}
          type="button"
        >
          Reset
        </button>
      </form>
    </div>
  </div>
</div>
`;
